app/inpage/ts/inpage.ts

```diff
@@ -284,12 +287,33 @@ class InterceptorMessageListener {
 
 	private readonly requestAccountsFromSigner = async (ask_eth_requestAccounts: boolean) => {
 		if (this.signerWindowEthereumRequest === undefined) return
-		const reply = await this.signerWindowEthereumRequest({ method: ask_eth_requestAccounts ? 'eth_requestAccounts' : 'eth_accounts', params: [] })
-		if (!Array.isArray(reply)) return
-		await this.sendMessageToBackgroundPage({ method: 'eth_accounts_reply', params: [reply, ask_eth_requestAccounts] })
-		if (this.waitForAccountsFromWallet !== undefined) {
-			this.waitForAccountsFromWallet.resolve(true)
-			this.waitForAccountsFromWallet = undefined
+		if (!ask_eth_requestAccounts) {
+			const reply = await this.signerWindowEthereumRequest({ method: 'eth_accounts', params: [] })
+			if (!Array.isArray(reply)) throw new Error('Signer returned something else than an array')
+			await this.sendMessageToBackgroundPage({ method: 'eth_accounts_reply', params: [{ type: 'success', accounts: this.signerAccounts, requestAccounts: ask_eth_requestAccounts }] })
+			if (this.waitForAccountsFromWallet !== undefined) {
+				this.waitForAccountsFromWallet.resolve(true)
+				this.waitForAccountsFromWallet = undefined
+			}
+			return
+		}
+		if (this.pendingSignerAddressRequest !== undefined) {
+			await this.pendingSignerAddressRequest
+			await this.sendMessageToBackgroundPage({ method: 'eth_accounts_reply', params: [{ type: 'success', accounts: this.signerAccounts, requestAccounts: ask_eth_requestAccounts }] })
```

Comment:
Does this need to be in a try/catch block like the one below?  Same question for the one above.